middle_end/flambda/closure_conversion.ml

```diff
@@ -601,13 +605,16 @@ and close_functions t external_env function_declarations : Flambda.named =
     let param_vars = List.map (Env.find_var closure_env) params in
     let params = List.map Parameter.wrap param_vars in
     let closure_bound_var = Function_decl.closure_bound_var decl in
+    (* The closure_bound_var for a declared function should
+       always have the debug info *)
+    assert (Option.is_some (Variable.debug_info closure_bound_var));
```

Comment:
This assertion should be moved to `Closure_origin.create` or maybe to `Closure_id.wrap`

Reply:
I moved the check inside `Function_decl.create`